src/polyglot-notebooks-vscode-common/src/vscodeNotebookManagement.ts

```diff
@@ -31,13 +31,44 @@ export async function replaceNotebookCellMetadata(notebookUri: vscode.Uri, cellI
 }
 
 export async function replaceNotebookMetadata(notebookUri: vscode.Uri, documentMetadata: { [key: string]: any }): Promise<boolean> {
```

Comment:
Do we have tests for this method? This was a pretty annoying bug and a significant code change.

If not, it might be a good conversation about how we can test things like this.

Reply:
no it needs to be refactored to be testable